rundeck-data-model/src/main/java/org/rundeck/app/data/providers/v1/ExecReportDataProvider.java

```diff
@@ -0,0 +1,28 @@
+package org.rundeck.app.data.providers.v1;
+
+import org.rundeck.app.data.model.v1.report.RdExecReport;
+import org.rundeck.app.data.model.v1.report.dto.SaveReportResponse;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public interface ExecReportDataProvider extends DataProvider{
+    RdExecReport get(Long id);
+    RdExecReport fromExecWithScheduledAndSave(Long executionId);
+    SaveReportResponse fromExecWithId(Long id);
+    SaveReportResponse saveFromMap(Map execReportMap);
```

Comment:
Please specify an interface that defines the data being sent instead of using a generic class.

This applies to all cases in the data provider. We want to avoid using generic structures such as maps in interfaces so future developers will be able to more quickly understand what is being transferred. When generic types such as maps are used it forces developers to go look at the code on both sides of the transfer to understand what is being done.